internal/gatewayapi/securitypolicy.go

```diff
@@ -220,22 +250,39 @@ func resolveSecurityPolicyRouteTargetRef(policy *egv1a1.SecurityPolicy, routes m
 	return route.RouteContext
 }
 
-func (t *Translator) translateSecurityPolicyForRoute(policy *egv1a1.SecurityPolicy, route RouteContext, xdsIR XdsIRMap) {
+func (t *Translator) translateSecurityPolicyForRoute(
+	policy *egv1a1.SecurityPolicy, route RouteContext,
+	resources *Resources, xdsIR XdsIRMap) error {
 	// Build IR
 	var (
 		cors *ir.CORS
 		jwt  *ir.JWT
+		oidc *ir.OIDC
+		err  error
 	)
 
 	if policy.Spec.CORS != nil {
-		cors = t.buildCORS(policy)
+		cors, err = t.buildCORS(policy.Spec.CORS)
+		if err != nil {
```

Comment:
I dont think we should return early, we should probably translate each field, and collect all `errs` and return them in the end

Reply:
Agree 👍